packages/matter.js/src/codec/BtpCodec.ts

```diff
@@ -0,0 +1,147 @@
+/**
+ * @license
+ * Copyright 2022-2023 Project CHIP Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ByteArray, Endian } from "../util/ByteArray.js";
+import { DataReader } from "../util/DataReader.js";
+import { DataWriter } from "../util/DataWriter.js";
+
+export interface BtpHandshakeRequest {
+    versions: number[],
+    attMtu: number,
+    clientWindowSize: number,
+}
+
+export interface BtpHandshakeResponse {
+    version: number,
+    attMtu: number,
+    windowSize: number
+}
+
+export interface BtpPacketPayload {
+    ackNumber?: number,
+    sequenceNumber: number,
+    msgLength?: number,
+    segmentPayload?: ByteArray
+}
+
+export interface BtpHeader {
+    handshakeBit: number,
+    managementBit: number,
+    ackMsgBit: number,
+    endSegmentBit: number,
+    beginSegmentBit: number
+}
+
+export interface BtpPacket {
+    header: BtpHeader,
+    payload: BtpPacketPayload
+}
+
+export const enum BtpHeaderBits {
+    HandshakeBit = 0b01000000,
+    ManagementMsg = 0b00100000,
+    AckMsg = 0b00001000,
+    EndSegment = 0b00000100,
+    BeginSegment = 0b00000001
+}
+
+export const enum BtpOpcode {
+    HandshakeManagementOpcode = 0x6c,
+}
+
+const HANDSHAKE_HEADER = 0b01100101;
+
+export class BtpCodec {
+
+    static decodeBtpHandshakeRequest(data: ByteArray): BtpHandshakeRequest {
+        const reader = new DataReader(data, Endian.Little);
+        return this.decodeRequestPayload(reader);
+    }
+
+    static decodeBtpPacket(data: ByteArray): BtpPacket {
+        const reader = new DataReader(data, Endian.Little);
+
+        return {
+            header: this.decodeHeader(reader),
+            payload: this.decodeBtpPacketPayload(reader)
+        };
+
+    }
+
+    static encodeBtpHandshakeResponse({ version, attMtu, windowSize }: BtpHandshakeResponse): ByteArray {
```

Comment:
Do we need that method at all? Just rename the other one :-)

Reply:
Done